hack/test-cmd.sh

```diff
@@ -257,6 +251,12 @@ echo "config files: ok"
 export KUBECONFIG="${HOME}/.kube/non-default-config"
 export CLUSTER_ADMIN_CONTEXT=$(oc config view --flatten -o template -t '{{index . "current-context"}}')
 
+pushd "$OS_ROOT" > /dev/null
+restore_nullglob=$(shopt -p nullglob)
```

Comment:
Sorry, that's fair enough—the commit message should have been more detailed.  I have updated the code with more comments and provided more detail in the commit message as to the problems with the old code.  The commit message also includes alternative implementations.  Here is what I put for the updated commit message:

The previous code is problematic for the following reasons:

• The intention of using find is to match recursively.  This intention is properly realized.

• The intention of piping the output of the `find` command into the `sort` command is presumably to produce a sorted list of filenames.  However, using the `find` command with the `-print0` flag and the `sort` command without the `-z`/`--zero-terminated` flag will result in  `sort`'s misinterpreting its input, generally interpreting it as a single line (if there are no newlines in the filenames printed by `find`), because `sort` does not know that records are separated by NUL bytes rather than LF bytes.  The `sort` command thus sees its input as a single record and does not actually sort its input as intended.

• The intention of using the `-print0` flag with `find` and the `-0` flag with `xargs` is presumably to handle filenames that contain whitespace.  However, the `xargs printf` command at the end of the pipeline outputs records separated by LF bytes, and we use this output in a shell substitution, so even if we fixed the problem with sort, we would still split filenames on whitespace when we assigned them to the tests variable.

It would be possible to rewrite the pipeline as follows to fix the first problem, so we would sort properly but then split incorrectly on assigning to tests:

```bash
find "${1}" -name '*.sh' -print0 | sort -zu | xargs -0 -n1 printf "%s\n"
```

Alternatively, we could forego whitespace-tolerance and simplify the entire command as follows:

```bash
find "${1}" -name '*.sh' | sort -u
```

As a further alternative, we could preserve the use of `find` and `sort` but use some more advanced Bash to avoid splitting on whitespace:

```bash
declare -a tests
while IFS= read -r -d '' filename
do tests+=("$filename")
done < <(find "${1}" -name '*.sh' -print0 | sort -zu)
```